src/main/java/org/folio/services/voucher/BatchVoucherGenerateService.java

```diff
@@ -136,6 +137,20 @@ private BatchedVoucher buildBatchedVoucher(Voucher voucher, Map<String, List<Vou
     batchedVoucher.setInvoiceNote(invoice.getNote());
     Organization organization = vendorsMap.get(invoice.getVendorId());
     batchedVoucher.setVendorName(organization.getName());
+    List<Adjustment> adjustments = new ArrayList<>();
+    List<InvoiceLine> invoiceLines = invoiceLinesMap.get(invoice.getId());
+    List<String> adjustmentIds = invoice.getAdjustments().stream()
+      .map(Adjustment::getId)
+      .collect(Collectors.toList());
+    if (invoiceLines != null && !invoiceLines.isEmpty()) {
+      for (InvoiceLine invoiceLine : invoiceLines) {
+        adjustments.addAll(invoiceLine.getAdjustments().stream()
+          .filter(adjustment -> adjustment.getId() == null && adjustment.getExportToAccounting() && !adjustmentIds.contains(adjustment.getAdjustmentId()))
```

Comment:
The same for condition in the filter

Reply:
Moved to getOnlyUniqueAndExportingAdjustments method